Include/internal/pycore_cell.h

```diff
@@ -0,0 +1,48 @@
+#ifndef Py_INTERNAL_CELL_H
+#define Py_INTERNAL_CELL_H
+
+#include "pycore_critical_section.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef Py_BUILD_CORE
+#  error "this header requires Py_BUILD_CORE define"
+#endif
+
+// Sets the cell contents to `value` and return previous contents. Steals a
+// reference to `value`.
+static inline PyObject *
+PyCell_SwapTakeRef(PyCellObject *cell, PyObject *value)
+{
+    PyObject *old_value;
+    Py_BEGIN_CRITICAL_SECTION(cell);
```

Comment:
Just checking so I understand correctly: this means GC triggered by another thread cannot happen in between these two sections right?

Reply:
`Py_BEGIN_CRITICAL_SECTION()` is not directly related to GC. It locks the per-object lock on `cell`. There's a more detail description in the file comments: 
https://github.com/python/cpython/blob/main/Include/internal/pycore_critical_section.h

However, you are right that a GC cannot happen between these two calls *in this case*. The two lines below can't trigger a GC, nor do they release the GIL (or "detach" in the free-threaded build), so another thread can't perform a GC either.